components/renku_data_services/storage/rclone_patches.py

```diff
@@ -0,0 +1,264 @@
+            "Provider": "",
+            "Default": "",
+            "Value": None,
+            "Examples": [
+                {
+                    "Value": "personal",
+                    "Help": (
+                        "Connect to your personal storage space. "
+                        "This data connector cannot be used to share access to a folder."
+                    ),
+                    "Provider": "",
+                },
+                {
+                    "Value": "shared",
+                    "Help": (
+                        "Connect a 'public' folder shared with others. "
+                        "A 'public' folder may or may not be protected with a password."
+                    ),
+                    "Provider": "",
+                },
+            ],
+            "Required": True,
+            "Type": "string",
+            "ShortOpt": "",
+            "Hide": 0,
+            "IsPassword": False,
+            "NoPrefix": False,
+            "Advanced": False,
+            "Exclusive": True,
+            "Sensitive": False,
+            "DefaultStr": "",
+            "ValueStr": "",
+        },
+        {
+            "Name": "public_link",
+            "Help": public_link_help,
+            "Provider": "shared",
+            "Default": "",
+            "Value": None,
+            "Examples": None,
+            "ShortOpt": "",
+            "Hide": 0,
+            "Required": True,
+            "IsPassword": False,
+            "NoPrefix": False,
+            "Advanced": False,
+            "Exclusive": False,
+            "Sensitive": False,
+            "DefaultStr": "",
+            "ValueStr": "",
+            "Type": "string",
+        },
+    ]
+    storage_copy["Options"].extend(custom_options)
+
+    # use provider to indicate if the option is for an personal o shared storage
+    for option in storage_copy["Options"]:
+        if option["Name"] == "url":
+            option.update({"Provider": "personal", "Default": url_value, "Required": False})
+        elif option["Name"] in ["bearer_token", "bearer_token_command", "headers", "user"]:
+            option["Provider"] = "personal"
+
+    # Remove obsolete options no longer applicable for Polybox or SwitchDrive
+    storage_copy["Options"] = [
+        o for o in storage_copy["Options"] if o["Name"] not in ["vendor", "nextcloud_chunk_size"]
+    ]
+
+    spec.append(storage_copy)
+    return spec
+
+
+def patch_polybox_storage(spec: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """Add polybox virtual storage that uses webdav."""
+    return add_webdav_based_storage(
+        spec,
+        prefix="polybox",
+        name="PolyBox",
+        description="Polybox",
+        url_value="https://polybox.ethz.ch/remote.php/webdav/",
+        public_link_help="Shared folder link. E.g., https://polybox.ethz.ch/index.php/s/8NffJ3rFyHaVyyy",
+    )
+
+
+def patch_switchdrive_storage(spec: list[dict[str, Any]]) -> list[dict[str, Any]]:
```

Comment:
suggestion: I think it does not hurt to make this function and all others non public and only make `apply_pathes` public. Just to make sure people are not trying to do funny things reusing these functions elsewhere. The suggestion below requires edits for the function wherever it is used in this file.

```suggestion
def __patch_switchdrive_storage(spec: list[dict[str, Any]]) -> list[dict[str, Any]]:
```